### PR TITLE
RSDK-4338 Only publish docs with release

### PR DESCRIPTION
**Pull request by maximpertsov**
Only update [ts.viam.dev](https://ts.viam.dev/) when we release a stable version, rather than on every merge

### CODE REVIEW

**Comment by cheukt** [comment]
would this also publish on release candidates?

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -89,7 +89,9 @@ jobs:
         uses: actions/upload-artifact@v3
         with:
           name: dist
-          path: viamrobotics-sdk-*.tgz
+          path: |
+            viamrobotics-sdk-*.tgz
+            docs/dist
```
Comment (maximpertsov) [inline .github/workflows/release.yml:94]: trying to remember - do we need to upload artifacts for any reason other than to break up the prepare and release jobs? if not maybe it's easier to combine them and remove all these artifact steps...

**Comment by maximpertsov** [comment]
> would this also publish on release candidates?

no - should publish docs there too?

File: .github/workflows/release.yml (hunk 1 above)
Comment (cheukt) [reply .github/workflows/release.yml:94]: I believe so, I think it's still worth separating the steps out tho

**Comment by cheukt** [comment]
na, sorry I misread

**Review by cheukt — APPROVED** [review]

**Pull request merged**